scripts/west_commands/blobs.py

```diff
@@ -0,0 +1,166 @@
+            FORMAT STRINGS
+            --------------
+
+            Blobs are listed using a Python 3 format string. Arguments
+            to the format string are accessed by name.
+
+            The default format string is:
+
+            "{default_fmt}"
+
+            The following arguments are available:
+
+            - module: name of the module that contains this blob
+            - abspath: blob absolute path
+            - status: short status (A: present, M: hash failure, D: not present)
+            - path: blob local path from <module>/zephyr/blobs/
+            - sha256: blob SHA256 hash in hex
+            - type: type of blob
+            - version: version string
+            - license_path: path to the license file for the blob
+            - uri: URI to the remote location of the blob
+            - description: blob text description
+            - doc-url: URL to the documentation for this blob
+            '''))
+
+        # Remember to update west-completion.bash if you add or remove
+        # flags
+        parser.add_argument('subcmd', nargs=1, choices=['list', 'fetch'],
+                            help='''Select the sub-command to execute.
+                            Currently only list and fetch are supported.''')
+
+        # Remember to update west-completion.bash if you add or remove
+        # flags
+        parser.add_argument('-f', '--format', default=default_fmt,
+                            help='''Format string to use to list each blob;
+                                    see FORMAT STRINGS below.''')
+
+        parser.add_argument('-m', '--modules', type=str, default="",
+                            help='''a comma-separated list of modules; only blobs whose
+                            names are on this list will be taken into account
+                            by the sub-command.''')
+        parser.add_argument('-a', '--all', action='store_true',
+                            help='use all modules.')
+ 
+        return parser
+
+    def get_status(self, path, sha256):
+        if not path.is_file():
+            return 'D'
+        with path.open('rb') as f: 
+            m = hashlib.sha256()
+            m.update(f.read())
+            if sha256.lower() == m.hexdigest():
+                return 'A'
+            else:
+                return 'M'
+
+    def get_blobs(self, args):
+        blobs = []
+        # From comma-separated to array
+        modules = args.modules.split(',')
+        for module in zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest):
+            mblobs = module.meta.get('blobs', None)
+            if not mblobs:
+                continue
+
+            # Filter by module
+            module_name = module.meta.get('name', None)
+            if not args.all and module_name not in modules:
+                continue
+
+            blobs_path = Path(module.project) / zephyr_module.MODULE_BLOBS_PATH
+            for blob in mblobs:
+                blob['module'] = module_name
```

Comment:
Values like 'module' and 'status' are used for indexing into blob repeatedly. Should there be a file-scoped variable that contains these names rather than using them directly?

Reply:
We could, but I just didn't think it was worth it, given the simplicity of the code. I would start with this and then, if complexity grows, create "macros" for them.